prisondersdilemma.py

```diff
@@ -0,0 +1,202 @@
+image_02 = ImageStim(win, image="im_02.jpg") 
+image_03 = ImageStim(win, image="im_03.jpg") 
+image_04 = ImageStim(win, image="im_04.jpg") 
+image_05 = ImageStim(win, image="im_05.jpg") 
+image_06 = ImageStim(win, image="im_06.jpg")
+image_07 = ImageStim(win, image="im_07.jpg")
+image_08 = ImageStim(win, image="im_08.jpg")
+image_09 = ImageStim(win, image="im_09.jpg")
+image_10 = ImageStim(win, image="im_10.jpg")
+
+
+opponent = Imagetim () #for later in trial loop maybe setIm (if that exists) for randomly assigning opponent based on condition
+
+# txt within game
+connecttxt = TextStim ()
+cooptxt = TextStim ()
+deftxt = TextStim ()
+
+#log file
+log = open ('logfileprisonersdilemma.txt', 'w')
+log.write(' .... ') #insert headers of what should be logged
+
+#response keys
+resp = getKeys(keyList = ('q', 'space', 'left', 'right'))
+
+strategy = [1,2,3,4,5]  #random assignment of opponents strategy
+shuffle(strategy)
+
+randostrategy = [1,2] #for strategy 3
+
+
+condlist = 
+n_trials =  #need to set that
+
+
+#starting the game
+background_image.draw()
+win.flip()
+
+
+#trial loop
+for i in range(n_trials):
+	
+	#strategies
+
+	#opponent always cooperates
+	if strategy == 1:
+		opponent.draw()
+		cooptxt.draw()
+		win.flip()
+		wait(4) #seconds
+		connecttxt.draw
+		#sentence draw 
+		win.flip()
+	
+	#opponent always defects
+	elif strategy == 2:
+		opponent.draw()
+		deftxt.draw()
+		win.flip()
+		wait(4) #seconds
+		connecttxt.draw
+		#sentence draw 
+		win.flip()
+	
+	#random coop or defect
+	elif strategy == 3:
+		shuffle(randostrategy)
+		if randostrategy == 1:
+			opponent.draw()
+			cooptxt.draw()
+			win.flip() 
+			wait(4)#seconds
+			#sentence draw 
+			win.flip()
+		elif: 
+			opponent.draw()
+			deftxt.draw()
+			win.flip()
+			wait(4) #seconds
+			connecttxt.draw
+			#sentence draw 
+			win.flip()
+	
+	elif strategy == 4:
+		if condlist[i] == 1:
+			opponent.draw()
+			cooptxt.draw()
+			win.flip() 
+			wait(4)#seconds
+			connecttxt.draw
+			#sentence draw 
+			win.flip()
+		elif condlist[i] > 1:
+			if resp == 'right':
+				opponent.draw()
+				cooptxt.draw()
+				win.flip() 
+				wait(4)#seconds
+				connecttxt.draw
+				#sentence draw 
+				win.flip()
+			elif resp == 'left':
+				opponent.draw()
+				deftxt.draw()
+				win.flip() 
+				wait(4)#seconds
+				connecttxt.draw
+				#sentence draw 
+				win.flip()
+	elif strategy == 5:
+		if condlist[i] == 1:
+			opponent.draw()
+			deftxt.draw()
+			win.flip() 
+			wait(4)#seconds
+			connecttxt.draw
+			#sentence draw 
+			win.flip()
+		elif condlist[i] > 1:
+			if resp == 'right':
+				opponent.draw()
+				cooptxt.draw()
+				win.flip() 
+				wait(4)#seconds
+				connecttxt.draw
+				#sentence draw 
+				win.flip()
+			elif resp == 'left':
+				opponent.draw()
+				deftxt.draw()
+				win.flip() 
+				wait(4)#seconds
+				connecttxt.draw
+				#sentence draw 
+				win.flip()
+				
+		
+			
```

Comment:
Once again I can see that your code with the different strategies does more or less the same thing: it checks whch strategy is being used, it draws on the displays, waits 4 sec and flips the screen. You can significantly shorten your code if you take some of these arguments into a function, so you don't need to repeat your operations on every if. I believe that will make your code more readable and easier to check for mistakes. 

Also, use 'elif' when there are more than 2 possibilities (for example: if 0 > number: do something;  elif 0<number <10: do something else; else: do a third thing). When you have only two possibilities (left and right) you can just use if and else, or even omit the else altogether (but be careful to position your code properly then).

Reply:
I agree! I think you already have most of it but it might be a good idea to develop general functions that take the strategy as an input argument and run the code in the if - statements. That would also make it more versatile! Other than that - well done so far!